EndlessClient/Dialogs/GuildDialog.cs

```diff
@@ -1,233 +1,522 @@
             _dialogIconService = dialogIconService;
             _localizedStringFinder = localizedStringFinder;
             _characterProvider = characterProvider;
             _messageBoxFactory = messageBoxFactory;
+            _textMultiInputDialogFactory = textMultiInputDialogFactory;
+            _contentProvider = contentProvider;
+            _guildActions = guildActions;
+            _guildSessionProvider = guildSessionProvider;
+            _characterInventoryProvider = characterInventoryProvider;
+            _eifFileProvider = eifFileProvider;
+            _stateStack = new Stack<State>();
+            _creationMemberCount = 0;
 
-            SetState(GuildDialogState.Initial);
+            SetState(new State(GuildDialogState.Initial));
 
             BackAction += (_, _) =>
             {
-                switch (_state)
+                if (_stateStack.Count > 0)
                 {
-                    case GuildDialogState.Management:
-                        SetState(GuildDialogState.Initial);
-                        break;
-                    default:
-                        // no-op
-                        break;
+                    var previousState = _stateStack.Pop();
+                    SetState(previousState);
+                }
+                else
+                {
+                    SetState(new State(GuildDialogState.Initial));
                 }
             };
 
             Title = _localizedStringFinder.GetString(EOResourceID.GUILD_GUILD_MASTER);
         }
 
-        private void SetState(GuildDialogState state)
+        protected override void OnUpdateControl(GameTime gameTime)
         {
-            if (state != GuildDialogState.Initial && _state == state)
-                return;
+            if (_state.DialogState == GuildDialogState.RegisterGuild)
+            {
+                _guildSessionProvider.CreationSession.MatchSome(creationSession =>
+                {
+                    if (creationSession.Approved)
+                    {
+                        SetState(new State(GuildDialogState.WaitingForMembers), false);
+                    }
+                });
+            }
+
+            if (_state.DialogState == GuildDialogState.WaitingForMembers)
+            {
+                _guildSessionProvider.CreationSession.MatchSome(creationSession =>
+                {
+                    if (creationSession.Members.Count != _creationMemberCount)
+                    {
+                        AddItemToList(new ListDialogItem(this, ListDialogItem.ListItemStyle.Small, 0)
+                        {
+                            PrimaryText = Capitalize(creationSession.Members.Last())
+                        }, false);
+                        _creationMemberCount += 1;
+                    }
+                });
+            }
 
-            _state = state;
+            base.OnUpdateControl(gameTime);
+        }
 
+        private void SetState(State newState, bool pushState = true)
+        {
```

Comment:
This method is now missing a guard against changing state if `_state == newState`